containers/src/Kokkos_DualView.hpp

```diff
@@ -283,6 +283,29 @@ class DualView : public ViewTraits< DataType , Arg1Type , Arg2Type, Arg3Type >
     }
   }
 
+  // given a View in HostSpace, create another view in the Default space and wrap them with a DualView
+  template<typename V>
+  DualView (V aView,
```

Comment:
Does `create_mirror_view(Space, ViewType v)` just return `v` if `Space::memory_space == ViewType::memory_space`?  If so, then you would just write one constructor that creates both `d_view` and `h_view` with `create_mirror_view`.

Reply:
More importantly, instead of using `std::enable_if`, you could set up the template parameters so that only a `Kokkos::View` works:
```c++
template<class ViewDataType, class ... ViewProperties>
DualView (Kokkos::View<ViewDataType, ViewProperties...> aView) :
  , d_view (create_mirror_view (typename t_dev::memory_space (), aView))
  , h_view (create_mirror_view (typename t_host::memory_space (), aView))
  /* ... modified_device and modified_host constructors ... */
{}
```

Reply:
Another question: if `t_dev::memory_space == CudaUVMSpace`, then `t_host::memory_space == CudaUVMSpace` as well.  This means that if you give a HostSpace View to the new DualView constructor, then the resulting DualView won't actually "view" the input View.  Is that what you want, or would you prefer to forbid this option at compile time?

Reply:
The thing Steve implemented currently would not work for `CudaUVMSpace` as the memory space of the `DualView`, because than the host view can't be constructed from a `HostSpace` view.

Reply:
@crtrott @mhoemmen right now `create_mirror_view` requires the memory spaces to be exactly the same. Should it, and should DualView, not make deep copies of views if the spaces are different but accessible?

Reply:
@ibaned To me, different spaces in `create_mirror_view` signify intent to create a new allocation.  For example, I could think of use cases for this:
```c++
View<double*, CudaSpace> x_cuda (...);
auto x_uvm = create_mirror_view (CudaUVMSpace (), x_cuda);
```